docs/src/main/asciidoc/cache.adoc

```diff
@@ -302,6 +297,18 @@ method annotated with `@CacheResult` or `@CacheInvalidate`.
 
 This annotation is optional and should only be used when some of the method arguments are NOT part of the cache key.
 
+== Caching programmatic API
+
+TODO
+
+=== CacheManager
+
+TODO
+
+=== Cache
```

Comment:
Provided we do injection, I winder if Cache should be the first one showed. Will depend on what's inside the todo

Reply:
Now that `Cache` can be injected using `@CacheName` (I'll detail that in another comment below), I agree it should come first in the doc so I moved it. The doc is still empty though :)

Reply:
This should be fixed with the latest commit.